### PR TITLE
Feature/contract details error

### PR DESCRIPTION
**Pull request by MathiasOhrstrom**
Item from design: []

### Checklist

- [x] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt [hunk 1]
```diff
@@ -74,11 +80,26 @@ class ContractDetailActivity : BaseActivity(R.layout.contract_detail_activity) {
...
+                    content.remove()
+                    error.root.apply {
+                        show()
+                        setBackgroundColor(context.colorAttr(R.attr.colorSurface))
+                    }
+                } else {
+                    content.show()
+                    error.root.remove()
+                    result.getOrNull()?.bindTo(binding.card)
+                    startPostponedEnterTransition()
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt:100]: We might need this row also when an error occurs - if we do not, there will be a large amount of delay upon attempting to navigate to this activity when it errors

**Review by hedvigoscar — APPROVED** [review]
Good solution!

Approved upon fixing comments

File: app/src/main/res/layout/generic_error.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
-    android:layout_height="wrap_content"
+    android:layout_height="match_parent"
```
[the remark this answers is not in the document]
Comment (MathiasOhrstrom) [reply app/src/main/res/layout/generic_error.xml:5]: It is so the color we apply to it does not cut off at the end of its content. And yes it still works as normal on the other screens

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt (hunk 1 above)
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt]: do you mean like this? 
```model.data.observe(this) { result ->
            binding.apply {
                if (result.isFailure) {
                    content.remove()
                    error.root.apply {
                        show()
                        setBackgroundColor(context.colorAttr(R.attr.colorSurface))
                    }
                } else {
                    content.show()
                    error.root.remove()
                    result.getOrNull()?.bindTo(binding.card)
                }
                startPostponedEnterTransition()
            }
        }
```

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt]: Yes!

**Pull request merged**